### PR TITLE
LPS-152789 Add spring context when using the portal api jar

### PR DESCRIPTION
**Pull request by Seiphon**

### CODE REVIEW

**Comment by Seiphon** [comment]
ci:test:sf

**Comment by Seiphon** [comment]
ci:test:sf

**Comment by Seiphon** [comment]
ci:test

**Comment by Seiphon** [comment]
ci:test:sf

**Comment by Seiphon** [comment]
ci:test:sf

**Comment by Seiphon** [comment]
these test failure caused by a ci outage. https://liferay.slack.com/archives/C19PWQ74J/p1651862539791659

File: modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/service/ServiceAnalyzerPlugin.java [hunk 1]
```diff
@@ -78,12 +79,20 @@ public boolean analyzeJar(Analyzer analyzer) throws Exception {
 				analyzer, "com.liferay.portal.spring.extender");
 		}
 
-		if (jar == null) {
+		Packages refferredPackages = analyzer.getReferred();
```
Comment (gamerson) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/service/ServiceAnalyzerPlugin.java:82]:
```suggestion
		Packages referredPackages = analyzer.getReferred();
```

File: modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/service/ServiceAnalyzerPlugin.java [hunk 2]
```diff
@@ -78,12 +79,20 @@ public boolean analyzeJar(Analyzer analyzer) throws Exception {
...
+
+		Object packages = refferredPackages.getByFQN(
```
Comment (gamerson) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/service/ServiceAnalyzerPlugin.java:84]:
```suggestion
		Attrs springExtenderServiceAttrs = refferredPackages.getByFQN(
```

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/springContext/build.gradle [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		}
+	}
+}
+
+task runGradleTest
+
+runGradleTest {
+	Project serviceProject = project(":modules:sample:sample-service")
+
+	dependsOn ":modules:sample:sample-service:buildService"
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/springContext/build.gradle:17]: Lets put the "dependsOn" line as the first thing after `runGradleTest`

**Review by gamerson — CHANGES_REQUESTED** [review]
Thanks Seiphon for this pull, just a few minor comments.

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/springContext/build.gradle (hunk 3 above)
Comment (Seiphon) [reply modules/sdk/gradle-plugins-workspace/src/gradleTest/springContext/build.gradle:17]: ok, thanks Greg, will resend the pr

**Comment by Seiphon** [comment]
have resent the pr here https://github.com/gamerson/liferay-portal/pull/539

**Pull request closed without merging**